### PR TITLE
add affinity to domain resource spec

### PR DESCRIPTION
**Pull request by CarolynRountree**
Fixes Jira #355

### CODE REVIEW

**Review by rakillen — APPROVED** [review]

File: webui/src/js/utils/k8s-domain-resource-generator.js [hunk 1]
```diff
@@ -247,7 +247,22 @@ define(['models/wkt-project', 'utils/k8s-domain-configmap-generator', 'js-yaml',
...
+      }
+
+      const resources = getKubernetesResourcesForCluster(cluster);
+      if (resources) {
+        serverPod.resources = resources;
+      }
+
+      serverPod.affinity = _getAffinityForServerPod(100);
+
+      return Object.keys(serverPod).length > 0 ? serverPod : null;
```
Comment (robertpatrick) [inline webui/src/js/utils/k8s-domain-resource-generator.js:265]: Why duplicate all the code from _getServerPod() just to add one line?  Why not just do this:

```
function getServerPodForCluster(cluster) {
    serverPod = _getServerPod(getJavaOptionsForCluster(cluster), getUserMemArgsForCluster(cluster), getKubernetesResourcesForCluster(cluster)) || {};
    serverPod.affinity = _getAffinityForServerPod(100);
    return serverPod;
}
```

File: webui/src/js/utils/k8s-domain-resource-generator.js [hunk 2]
```diff
@@ -353,6 +369,21 @@ define(['models/wkt-project', 'utils/k8s-domain-configmap-generator', 'js-yaml',
...
+      const keyObject = {'key': 'weblogic.cluster.name', 'operator':'In', 'values':['$(CLUSTER_NAME)']};
+      podAffinityTerm.labelSelector.matchExpressions = [keyObject];
+      podAffinityTerm.topologyKey = 'kubernetes.io/hostname';
+      const podObject = {'weight': weight, podAffinityTerm};
+
+      const preferredDuringSchedulingIgnoredDuringExecution = [podObject];
+      const preferred = {preferredDuringSchedulingIgnoredDuringExecution};
+      return  {'podAntiAffinity': preferred};
+    }
+
```
Comment (robertpatrick) [inline webui/src/js/utils/k8s-domain-resource-generator.js:384]: This seems like a lot of work for returning a predefined object.  There is no need to build the object up like in Java so why not just do something like:

```
return {
  podAntiAffinity: {
    preferredDuringSchedulingIgnoredDuringExecution: [{
      weight: 100
      podAffinityTerm: {
        topologyKey: 'kubernetes.io/hostname'
        labelSelector: {
          matchExpressions: [{
            key: 'weblogic.clusterName'
            operator: 'In'
            values: [{ '$(CLUSTER_NAME)' }]
          }]
        }
      }
    }]
  }
};
```

**Review by robertpatrick — CHANGES_REQUESTED** [review]

File: webui/src/js/utils/k8s-domain-resource-generator.js (hunk 1 above)
Comment (CarolynRountree) [reply webui/src/js/utils/k8s-domain-resource-generator.js:265]: Server Pod might be empty, and I did not know how to add const serverPod = {}

Comment (CarolynRountree) [reply webui/src/js/utils/k8s-domain-resource-generator.js:257]: resolved code per suggestions

File: webui/src/js/utils/k8s-domain-resource-generator.js (hunk 2 above)
Comment (CarolynRountree) [reply webui/src/js/utils/k8s-domain-resource-generator.js:383]: reworked code per above

**Review by robertpatrick — APPROVED** [review]

**Pull request merged**